chat/src/main/java/com/github/twitch4j/chat/TwitchChat.java

```diff
@@ -327,6 +339,17 @@ public TwitchChat(EventManager eventManager, CredentialManager credentialManager
 
         // register event handler
         eventManager.onEvent(ChannelMessageEvent.class, this::onChannelMessage);
+        eventManager.onEvent(IRCMessageEvent.class, event -> {
+            // we get at least one room state event with channel name + id when we join a channel, so we cache that to provide channel id + name for all events
+            if (event.getCommandType().equals("ROOMSTATE")) {
+                if (event.getChannelId() != null && !channelIdToChannelName.containsKey(event.getChannelId())) {
```

Comment:
I might not be so strict on a contains check here; if a name change occurs, for example, we could update the cache to be consistent with the latest name